### PR TITLE
Add support for filtering PredictionSurv objects

### PR DESCRIPTION
**Pull request by mllg**

### CODE REVIEW

**Comment by mllg** [comment]
Not sure about field `distr` ... is this now always a matrix?

**Comment by mllg** [comment]
Oh, and this PR requires mlr3 v0.13.0 ...

File: R/PredictionDataSurv.R [hunk 1]
```diff
@@ -87,3 +87,24 @@ c.PredictionDataSurv = function(..., keep_duplicates = TRUE) {
...
+  if (!is.null(pdata$crank)) {
+    pdata$crank = pdata$crank[keep]
+  }
+
+  if (!is.null(pdata$lp)) {
+    pdata$lp = pdata$lp[keep]
+  }
+
+  if (!is.null(pdata$distr)) {
+    pdata$distr = pdata$distr[keep,, drop = FALSE]
```
Comment (RaphaelS1) [inline R/PredictionDataSurv.R:106]:
```suggestion
if (is.matrix(pdata$distr)) {
    pdata$distr = pdata$distr[keep,, drop = FALSE]
} else {
    pdata$distr = pdata$distr[keep]
}
```

**Comment by RaphaelS1** [comment]
Sorry @mllg only just saw this.

> Not sure about field distr ... is this now always a matrix?

No it can be a matrix or `distr6::Distribution`, I've added a suggestion above to fix this

**Pull request merged**